### PR TITLE
Get binding generation & publishing ready for 0.1 release

### LINKED ISSUE #73: Document MSRV guarantees and enforce in CI

### PR DESCRIPTION
**Pull request by tnull**
~~Based on #124.~~
Follow-up to #59.
Closes #73.

Here we'll add the final necessary steps to get bindings ready for the initial 'official' release.

### CODE REVIEW

File: bindings/kotlin/ldk-node-android/lib/src/androidTest/kotlin/org/lightningdevkit/ldknode/AndroidLibTest.kt [hunk 1]
```diff
@@ -16,9 +16,9 @@ import androidx.test.ext.junit.runners.AndroidJUnit4
 @RunWith(AndroidJUnit4::class)
 class AndroidLibTest {
     @Test fun node_start_stop() {
-        val network: Network = "regtest"
-        assertEquals(network, "regtest")
+        val network = Network.REGTEST
 
+        val esplora_url = "http://127.0.0.1:3002"
```
Comment (jkczyz) [inline bindings/kotlin/ldk-node-android/lib/src/androidTest/kotlin/org/lightningdevkit/ldknode/AndroidLibTest.kt:21]: Changes look unrelated to the rest of the commit. Also, `esplora_url` is not used.

File: Package.swift [hunk 2]
```diff
@@ -0,0 +1,35 @@
+// swift-tools-version:5.5
+// The swift-tools-version declares the minimum version of Swift required to build this package.
+
+import PackageDescription
+
+let tag = "0.1-alpha.1"
```
Comment (jurvis) [inline Package.swift:6]: side note: I don't think the `-alpha.1` suffix is supported by SwiftPM when we tag the release last I checked

File: Package.swift [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+            name: "LDKNode",
+            dependencies: ["LDKNodeFFI"],
+            path: "./bindings/swift/Sources",
+            swiftSettings: [.unsafeFlags(["-suppress-warnings"])]
+        )
+        .binaryTarget(
+            name: "LDKNodeFFI",
+            url: url,
+            checksum: checksum,
+            ),
```
Comment (jurvis) [inline Package.swift:33]: nit: stray tab

File: bindings/swift/LDKNodeFFI.xcframework/Info.plist [hunk 4]
```diff
@@ -8,7 +8,7 @@
 			<key>LibraryIdentifier</key>
```
Comment (jurvis) [inline bindings/swift/LDKNodeFFI.xcframework/Info.plist:8]: q: what are we using the checked in `.xcframework` for?

File: scripts/uniffi_bindgen_generate_kotlin_android.sh [hunk 5]
```diff
@@ -3,15 +3,15 @@ BINDINGS_DIR="bindings/kotlin"
 TARGET_DIR="target"
 PROJECT_DIR="ldk-node-android"
 PACKAGE_DIR="org/lightningdevkit/ldknode"
-UNIFFI_BINDGEN_BIN="cargo +nightly run --features=uniffi/cli --bin uniffi-bindgen"
+UNIFFI_BINDGEN_BIN="cargo run --features=uniffi/cli --bin uniffi-bindgen"
```
Comment (jurvis) [inline scripts/uniffi_bindgen_generate_kotlin_android.sh:6]: nice!

**Review by jurvis — APPROVED** [review]
took a glance and looks great to me + had some questions :)

File: scripts/swift_update_package_checksum.py [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+		if new_value is None:
+			continue
+
+		print(f'setting {current_variable_name} (JSON-serialization):')
+		print(json.dumps(new_value))
+
+		regex = re.compile(f'(let[\s]+{current_variable_name}[\s]*=[\s]*)(.*)')
+
+		previous_value = regex.search(package_file).group(2)
+		# new_package_file = checksum_regex.sub(f'\g<1>"{new_checksum}"', package_file)
```
Comment (jkczyz) [inline scripts/swift_update_package_checksum.py:68]: Is this needed?

File: bindings/swift/LDKNodeFFI.xcframework/Info.plist (hunk 4 above)
Comment (reez) [reply bindings/swift/LDKNodeFFI.xcframework/Info.plist:8]: > Mh, we're not checking in the full framework, just the necessary umbrella headers. Correct me if I'm wrong, but my understanding is this is necessary so that it can be used as a local dependency, without SwiftPM? (cc @reez)

Yeah this is similar to how BDK swift ffi is set up related to how they allow for local import of a Swift Package

**Review by jkczyz — COMMENTED** [review]
LGTM, but I'll let @jurvis speak to the Swift changes. Feel free to squash the fixups. IMO.

File: bindings/swift/LDKNodeFFI.xcframework/Info.plist (hunk 4 above)
Comment (jurvis) [reply bindings/swift/LDKNodeFFI.xcframework/Info.plist:8]: 👍 that makes sense

File: scripts/swift_update_package_checksum.py [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+	original_package_file = None
+	try:
+		with open(package_file_path, 'r') as package_file_handle:
+			original_package_file = package_file_handle.read()
+	except:
+		print('Failed to read Package.swift file.', file=sys.stderr)
+		sys.exit(1)
+
+	package_file = original_package_file
```
Comment (jkczyz) [inline scripts/swift_update_package_checksum.py:55]: Is the additional variable needed? @arik-so Do you have any insight?

**Review by jkczyz — APPROVED** [review]
